### PR TITLE
fix: replace usage of `pkg_resources`

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

fixes: #1995

### How I did it

use importlib distributions

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Comment by antazoey** [comment]
found a better way! in coming, actually way better

File: setup.py [hunk 1]
```diff
@@ -97,7 +97,6 @@
     install_requires=[
         "click>=8.1.6,<9",
         "ijson>=3.1.4,<4",
-        "importlib-metadata",
```
Comment (fubuloubu) [inline setup.py:100]: this is bundled python 3.8+?

**Review by fubuloubu — APPROVED** [review]

File: setup.py (hunk 1 above)
Comment (antazoey) [reply setup.py:100]: Yes, in 3.8+, you can now use `importlib.metadata` directly, so we could have done this a while ago

Comment (antazoey) [reply setup.py:100]: Ugh - I guess in 3.8 you can't select by group.
Whyyy Python whyy.

**Comment by antazoey** [comment]
I made sure plugins installed from remote, editable mode, and regular pip look good in `ape plugins list`

**Pull request merged**